pkg/addon/addon.go

```diff
@@ -362,26 +363,60 @@ func RenderApplication(addon *types.Addon, args map[string]string) (*v1beta1.App
 	for _, tmpl := range addon.YAMLTemplates {
 		comp, err := renderRawComponent(tmpl)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		app.Spec.Components = append(app.Spec.Components, *comp)
 	}
 	for _, tmpl := range addon.CUETemplates {
 		comp, err := renderCUETemplate(tmpl, addon.Parameters, args)
 		if err != nil {
-			return nil, ErrRenderCueTmpl
+			return nil, nil, ErrRenderCueTmpl
 		}
 		app.Spec.Components = append(app.Spec.Components, *comp)
 	}
-	for _, def := range addon.Definitions {
-		comp, err := renderRawComponent(def)
-		if err != nil {
-			return nil, err
+
+	var defObjs []*unstructured.Unstructured
+
+	if isDeployToRuntimeOnly(addon) {
+		// Runtime cluster mode needs to deploy definitions to control plane k8s.
+		for _, def := range addon.Definitions {
+			obj, err := renderObject(def)
+			if err != nil {
+				return nil, nil, err
+			}
+			defObjs = append(defObjs, obj)
+		}
+	} else {
+		for _, def := range addon.Definitions {
+			comp, err := renderRawComponent(def)
+			if err != nil {
+				return nil, nil, err
+			}
+			app.Spec.Components = append(app.Spec.Components, *comp)
 		}
-		app.Spec.Components = append(app.Spec.Components, *comp)
 	}
 
-	return app, nil
+	return app, defObjs, nil
+}
+
+func isDeployToRuntimeOnly(addon *types.Addon) bool {
```

Comment:
Why is `isDeployToRuntimeOnly` , why can't flux be installed both runtime and control cluster?

Reply:
Yes. It will..
The runtime clusters include the control plane.